lib/remote/configstageshandler.cpp

```diff
@@ -128,19 +136,18 @@ void ConfigStagesHandler::HandlePost(
 		if (reload && !activate)
 			BOOST_THROW_EXCEPTION(std::invalid_argument("Parameter 'reload' must be false when 'activate' is false."));
 
-		std::unique_lock<std::mutex> lock(ConfigPackageUtility::GetStaticPackageMutex());
-
 		stageName = ConfigPackageUtility::CreateStage(packageName, files);
 
 		/* validate the config. on success, activate stage and reload */
-		ConfigPackageUtility::AsyncTryActivateStage(packageName, stageName, activate, reload);
+		ConfigPackageUtility::TryActivateStage(packageName, stageName, activate, reload);
```

Comment:
So the HTTP request uploading the new configuration will now wait for the complete config validation to complete? I don't think this is a good idea as this may cause timeouts in existing clients as this could take quite long.

Reply:
Oh 🐘 , I hadn't thought of that.